### PR TITLE
Feature/add new category

### LINKED ISSUE #29: Rating functionality for a category
Idea for first step: Submit a rating when adding a new evaluation
## Value proposition
**As a** user
**I need** to rate the pupil in the submitted category
**to be able to** choose an evaluation according to this evaluation.

## Description
Bildschirmfoto 2021-12-06 um 09 36 13

Bildschirmfoto 2021-12-06 um 09 48 20

## Acceptance criteria
- [x] A new category can only be added if a rating is submitted as well.
- [x] Possible evaluations of a category are shown dependent on the chosen rating.
- [x] The evaluations use the first name of the selected pupil.
- [x] If there is no evaluation availabe, a note and button is shown to add a new one (via AddCategory Form); after submitting a new evaluation, it should be displayed at the last view of the "AddEvaluationForm".
- [x] After clicking on an evaluation, it should be highlighted (bold) so that the user knows which one he chose.

## Tasks

- [x] Refactor AddEvaluation: (1) rename "AddCategory"; (2) add rating to be saved with evaluation
- [x] Add localStorage item "Evaluations" with four default values ("Arbeiten in der Gruppe", "Arbeitsweise", "Konfliktverhalten", "Verhalten gegenüber dem Lehrer") and two default evaluations per rating
- [x] Add rating functionality with 5 options (1, 2, 3, 4, 5)  [idea here](https://reactjsexample.com/a-simple-star-rating-component-with-react/)
- [x] Refactor AddPupilForm: only the pupil can be added
- [x] Add "AddEvaluationForm" with four steps: (1) Choose a category, (2) choose a rating, (3) choose a sentence, (4) add/save to pupil's evaluations
- [x] Add different active colour to button when a category is chosen

### PR DESCRIPTION
**Pull request by PReichetanz**
Implement NewEvaluationForm into App

- You can add a new evaluation at two points: when in ClassOverview via "Einstellungen" or if there is no evaluation for your chosen category and rating.
Bildschirmfoto 2021-12-08 um 12 52 47
Bildschirmfoto 2021-12-08 um 12 52 57

[View at heroku](https://capstone-rev-feature-ad-t4hnva.herokuapp.com/)

Closes #29

### CODE REVIEW

File: src/app/components/AddEvaluationForm/AddEvaluationForm.tsx [hunk 1]
```diff
@@ -18,11 +19,14 @@ export default function AddEvaluationForm({
...
 }: AddEvaluationFormProps): JSX.Element {
-  const { evaluations } = useEvaluations();
+  const { evaluations, addNewEvaluation } = useEvaluations();
+  const [isNewEvaluationFormShown, setIsNewEvaluationFormShown] =
+    useState(false);
   const [selectedCategory, setSelectedCategory] = useState('');
   const [selectedEvaluation, setSelectedEvaluation] = useState('');
   const [selectedRating, setSelectedRating] = useState(0);
   const [inputError, setInputError] = useState(false);
+  console.log('selectedRating in AddEvaluationForm ' + selectedRating);
```
Comment (lajanzen) [inline src/app/components/AddEvaluationForm/AddEvaluationForm.tsx]: Do you still need this? 

```suggestion
```

File: src/app/components/AddEvaluationForm/AddEvaluationForm.tsx [hunk 2]
```diff
@@ -92,8 +105,9 @@ export default function AddEvaluationForm({
           evaluations.map((evaluation) =>
```
Comment (lajanzen) [inline src/app/components/AddEvaluationForm/AddEvaluationForm.tsx:104]: The whole part of the function from here looks very nested with many conditions, is there a way you could make the code a little clearer to read?

File: src/app/pages/ClassOverview/ClassOverview.tsx [hunk 3]
```diff
@@ -1,23 +1,39 @@
...
   }
 
+  function handleNewCategory(
+    category: string,
+    rating: number,
+    evaluation: string
+  ) {
+    console.log(category);
+    console.log(rating);
+    console.log(evaluation);
```
Comment (lajanzen) [inline src/app/pages/ClassOverview/ClassOverview.tsx]:
```suggestion
```

File: src/app/pages/ClassOverview/ClassOverview.tsx [hunk 4]
```diff
@@ -1,23 +1,39 @@
...
+    addNewEvaluation(category, rating, evaluation);
+    setIsSettingShown(false);
+  }
+
   function getTextToCopy(pupils: Pupil[] | undefined) {
     let textToCopy = '';
```
Comment (lajanzen) [inline src/app/pages/ClassOverview/ClassOverview.tsx:35]:
```suggestion
    let textToCopy;
```

File: src/app/components/NewEvaluationForm/NewEvaluationForm.stories.tsx [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import AddCategoryForm from './AddCategoryForm';
+import AddCategoryForm from './NewEvaluationForm';
```
Comment (janhendrikstruve) [inline src/app/components/NewEvaluationForm/NewEvaluationForm.stories.tsx:2]: Good Naming! 👍

File: src/app/pages/ClassOverview/ClassOverview.tsx [hunk 6]
```diff
@@ -57,23 +73,27 @@ export default function ClassOverview(): JSX.Element {
             onCancel={() => setIsFormShown(false)}
           />
         )}
+        {isSettingShown && (
+          <NewEvaluationForm
+            onSubmit={handleNewCategory}
+            onCancel={() => setIsSettingShown(false)}
+            missingInput={false}
```
Comment (janhendrikstruve) [inline src/app/pages/ClassOverview/ClassOverview.tsx]: Isn't "inputFilled" and setting it true a less confusing variant?

File: src/app/pages/ClassOverview/ClassOverview.tsx [hunk 7]
```diff
@@ -57,23 +73,27 @@ export default function ClassOverview(): JSX.Element {
...
         {pupils.map((pupil) => (
           <Card key={pupil.id} pupil={pupil} deleteCard={deletePupil} />
         ))}
         {pupils.length === 0 ? '' : <CopyButton copyText={dataToCopy} />}
       </Main>
-      <AddButton onClick={() => setIsFormShown(true)}>
-        Schüler hinzufügen
-      </AddButton>
+      <Navigation
+        isFormNavigation={false}
```
Comment (janhendrikstruve) [inline src/app/pages/ClassOverview/ClassOverview.tsx:86]: Maybe name it something like formNavigtionStatus?

File: src/app/pages/PupilOverview/PupilOverview.tsx [hunk 8]
```diff
@@ -19,7 +20,7 @@ export default function PupilOverview(): JSX.Element {
 
   function handleFormSubmit(category: string, evaluation: string) {
     addEvaluation(currentPupil, category, evaluation);
-    setIsFormShown(false);
+    setIsAddEvaluationFormShown(false);
```
Comment (janhendrikstruve) [inline src/app/pages/PupilOverview/PupilOverview.tsx:23]: I would also consider something like AddEvaluationFormShowStatus or AddEvaluationFormShowIndictor, but that's all down to personal preferences is guess :)

**Review by janhendrikstruve — COMMENTED** [review]
Very nice code! I have some Ideas for some naming, maybe that's helpful to you :)

**Pull request merged**